src/lib/coda_transition/external_transition.ml

```diff
@@ -306,6 +395,30 @@ module Make
     | Error e ->
         Error (`Verifier_error e)
 
+  let validate_delta_transition_chain_witness (t, validation) =
+    (* I didn't use the Transition_chain_witness.verify function because otherwise it
+       would include a cyclic dependencies *)
```

Comment:
This could be done if we separate the `proof` and `verify` in two separate lib.

`proof` needs to know about transition_frontier, but `verify` don't.

Reply:
I would argue strongly for pulling it out. Having two implementations of the verification function that are supposed to be the same is bad.